File: easy/strings/reverse-integer.js
const reverse = x =>  {
    let num = x.toString().split("").reverse().join("")
    if(num[num.length - 1] === "-") {
        num = parseInt(num.slice(0, num.length - 1))
        if(num > 2 ** 31) return 0
        return parseInt(`-${num}`)
    }
    num = parseInt(num)
    if(num >= 2 ** 31) return 0
    return num
};

console.log(reverse(123))
console.log(reverse(-123))
console.log(reverse(120))
console.log(reverse(-120))
console.log(reverse(2147483648))
console.log(reverse(-2147483648))

File: easy/strings/reverse-integer.py
def reverse(x):
  num = str(x)[::-1]
  if num[-1] == "-":
    return 0 if int(num[:-1]) >= 2 ** 31 else int(f"-{int(num[:-1])}")
  return 0 if int(num) > 2 ** 31 - 1 else int(num)

print(reverse(123))
print(reverse(-123))
print(reverse(120))
print(reverse(-120))
print(reverse(2147483648))
print(reverse(-2147483648))

a = [1,2,3,4,5]
# a[start:stop:step]
print(a[1:4:2])
# => [2,4]